src/hotspot/share/compiler/compilerDirectives.hpp

```diff
@@ -185,6 +185,37 @@ class ControlIntrinsicIter {
   ControlIntrinsicIter& operator++();
 };
 
+class ControlIntrinsicValidator {
+ private:
+  bool _valid;
+  char* _bad;
+
+ public:
+  ControlIntrinsicValidator(ccstrlist option, bool disabled_all) : _valid(true), _bad(NULL) {
+    for (ControlIntrinsicIter iter(option, disabled_all); *iter != NULL && _valid; ++iter) {
+      if (vmIntrinsics::_none == vmIntrinsics::find_id(*iter)) {
+        _bad = NEW_C_HEAP_ARRAY(char, strlen(*iter) + 1, mtCompiler);
```

Comment:
Hi, 

I wanted something like this:

        size_t len = MIN(strlen(*iter), 64);        // cap len to a value we know is enough for all intrinsic names
        _bad = NEW_C_HEAP_ARRAY(char, len + 1, mtCompiler);
        strncpy(_bad, *iter, len + 1);                 // use strnlen to cap copy length

Reply:
okay. I see.  I make change to ensure the _bad is a bound string. here is a test. the _bad is limited to 64  even user-input is overflown. 
```
java -XX:+UnlockDiagnosticVMOptions -XX:ControlIntrinsic=+_dtan,+_0123456789012345678901234567890123456789012345678901234567890123456789 -version
Unrecognized intrinsic detected in ControlIntrinsic: _012345678901234567890123456789012345678901234567890123456789012
```

Reply:
limited the bound string 64 bytes instead of 65 in case it takes another cacheline.